src/v/transform/rpc/client.h

```diff
@@ -84,12 +84,17 @@ class client {
 
     ss::future<model::cluster_transform_report> generate_report();
 
+    ss::future<cluster::errc> create_transform_logs_topic();
```

Comment:
Instead of exposing this and the topic config (which feels a little bit of a breech of keeping this responsible for a single thing), what if we just wrapped `produce`? It feels a bit more in line with other methods in this class.

For example:

```
ss::future<cluster::errc> produce_logs(ss::chunked_fifo<model::record_batch>);
```

That method:

Would auto create the topic if not found, and otherwise call `produce` with the right topic_partition.

Reply:
I guess the other consideration is that we want to compute the topic partition before calling produce. I'll give this some thought though; I haven't really considered any of it since changing the design.

Reply:
Ah right, yeah. Anyways take a think, I feel like there are a few layers here that could be corrected.

Alternatively, just exposing the create topic, but not topic configuration from the client feels OK, but it would probably be better to pass in `topic_creator` to the transform logging sink.

Maybe it makes sense to have an interface like this:

```c++
class transform::logging::sink {
  virtual ss::future<> initialize() = 0;
  virtual ss::future<errc> emit(model::partition_id, io::json_batches) = 0;
  virtual result<model::partition_id, errc> compute_partition(model::transform_name_view) = 0;
  
};
```

And then have some code (a bool?) in the manager to lazily initialize the sink. And initialize just calls try_create_topic

Reply:
LGTM